### PR TITLE
feat: move onResolversCalled to the top level

### PR DESCRIPTION
**Pull request by n1ru4l**
Important performance change.

### CODE REVIEW

**Review by dotansimha — APPROVED** [review]

**Review by kamilkisiela — APPROVED** [review]

File: .changeset/fresh-toes-hang.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+---
+'@envelop/core': major
+---
+
+Move `onResolversCalled` from within `OnExecuteHookResult` and `OnSubscribeHookResult` to the `Plugin` type.
+
+```diff
+import type { Plugin } from "@envlop/core";
```
Comment (n1ru4l) [inline .changeset/fresh-toes-hang.md]:
```suggestion
import type { Plugin } from "@envelop/core";
```

File: .changeset/fresh-toes-hang.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+```
+
+We highly recommend avoiding to use any plugins that use `onResolversCalled` within your production environment as it has severe impact on the performance of the individual resolver functions within your schema.
+
+The schema resolver functions are now ONLY wrapped if any plugin in your envelop setup uses the `onResolversCalled` hook.
+
+If you need any shared state between `onExecute` and `onResolversCalled` you can share it by extending the context object.
+
+```ts
+import type { Plugin } from '@envlop/core';
```
Comment (n1ru4l) [inline .changeset/fresh-toes-hang.md]:
```suggestion
import type { Plugin } from '@envelop/core';
```

**Review by saihaj — CHANGES_REQUESTED** [review]
need to update site https://www.envelop.dev/docs/plugins/lifecycle#plugins-api

**Comment by n1ru4l** [comment]
@saihaj done 🥳

**Review by saihaj — APPROVED** [review]

**Pull request merged**